Data_Structures_Answers.md

```diff
@@ -2,17 +2,78 @@ Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
 
+
+#  def append(self, item):
+
+#    try:
+#        self.storage[self.current] = item -- O(1)
+#        self.current += 1 -- O(1)
+#
+#    except:
+#      self.current = 0 -- O(1)
+#      self.append(item) -- O(1)
+
+The time complexity seems to simplify to O(1)
+
 2. What is the space complexity of your ring buffer's `append` function?
 
+#  def append(self, item):
+
+#    try:
+#        self.storage[self.current] = item -- O(1)
+#        self.current += 1 -- O(1)
+#
+#    except:
+#      self.current = 0 -- O(1)
+#      self.append(item) -- O(1)
+
+The space complexity seems to simplify to O(1)... this is because at any given function call it is primarily using memory for two items ... the storage variable and the current variable.
```

Comment:
You are correct but not necessarily for the right reason. It's O(1) because when we are appending we aren't using any additional memory